website/addons/base/views.py

```diff
@@ -466,20 +503,41 @@ def addon_view_or_download_file_legacy(**kwargs):
 
 @must_be_valid_project
 @must_be_contributor_or_public
-def addon_deleted_file(auth, node, **kwargs):
+def addon_deleted_file(auth, node, error_type='BLAME_PROVIDER', **kwargs):
     """Shows a nice error message to users when they try to view
     a deleted file
     """
     # Allow file_node to be passed in so other views can delegate to this one
-    trashed = kwargs.get('file_node') or TrashedFileNode.load(kwargs.get('trashed_id'))
-    if not trashed:
-        raise HTTPError(httplib.NOT_FOUND, {
-            'message_short': 'Not Found',
-            'message_long': 'This file does not exist'
-        })
+    file_node = kwargs.get('file_node') or TrashedFileNode.load(kwargs.get('trashed_id'))
+    deleted_by, deleted_on = None, None
+    if isinstance(file_node, TrashedFileNode):
+        # TODO test for None once hotfix is in
+        print('attrs are ' + str(dir(file_node)))
+        deleted_by = str(file_node.deleted_by)
+        deleted_on = str(file_node.deleted_on)
+        if deleted_by == "None":
```

Comment:
`error_type = 'BLAME_PROVIDER' if file_node.deleted_by is None else 'FILE_GONE'`